src/System.Slices/System/ReadOnlySpan.cs

```diff
@@ -143,6 +143,16 @@ internal ReadOnlySpan(object obj, UIntPtr offset, int length)
             return new ReadOnlySpan<T>(slice.Object, slice.Offset, slice.Length);
         }
 
+        /// <summary>
+        /// Gets ArraySegment containing the ReadOnlySpan's data.
+        /// </summary>
+        /// <remarks>This method can be used for interop, as array data is mutable, an array is allocated and copy of the data is returned in it.</remarks>
+        public static explicit operator ArraySegment<T>(ReadOnlySpan<T> span)
```

Comment:
CreateArraySegment could also be an extension method, which I like, as I am not super thrilled by Span<T> now depending on ArraySegment<T>. I would prefer to keep Span<T> as simple as language primitives.